File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.py
class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:
        d = {}
        same = 0
        for i in range(len(grid)):
            s = str(grid[i])
            if s in d:
                d[s] += 1
            else:
                d[s] = 1
        for j in range(len(grid[0])):
            temp = []
            for i in range(len(grid)):
                temp.append(grid[i][j])
            temp = str(temp)
            if temp in d:
                same += d[temp]
        return same
                
        
        

File: 2352-equal-row-and-column-pairs/2352-equal-row-and-column-pairs.java
class Solution {
    public int equalPairs(int[][] grid) {
        
        int count = 0;
        int n = grid.length;
        
        Map<String, Integer> d = new HashMap<>();
        for(int [] row : grid){
            String rowString = Arrays.toString(row);
            d.put(rowString, 1 + d.getOrDefault(rowString, 0));
        }
        
        for(int c = 0; c < n; c++){
            int [] colArray = new int[n];
            for (int r = 0; r < n; r ++){
                colArray[r] = grid[r][c];
            }
            String colString = Arrays.toString(colArray);
            count += d.getOrDefault(colString, 0);
        }
        
        return count;
        
    }
}